tests/scenario/test_charm.py

```diff
@@ -0,0 +1,601 @@
+    # and end up with an blocked status.
+    relation = scenario.PeerRelation("takahe-peer", "takahe-peer")
+    state = scenario.State(leader=False, relations=[relation])
+    out = ctx.run(relation.created_event, state)
+    assert out.unit_status.name == "blocked"
+
+
+def test_peer_created_leader(ctx):
+    relation = scenario.PeerRelation("takahe-peer", "takahe-peer")
+    state = scenario.State(leader=True, relations=[relation])
+    out = ctx.run(relation.created_event, state)
+    # At this point we are still blocked, waiting for the config.
+    assert out.unit_status.name == "blocked"
+    assert len(out.secrets) == 1
+    # SCENARIO-NOTE: The [0] here is the revision, I think, which seems ugly.
+    # Also, why a dict and not a list if this is the way?
+    assert "takahe-secret-key" in out.secrets[0].contents[0]
+    assert len(out.secrets[0].contents[0]["takahe-secret-key"]) == 128
+    assert out.relations[0].local_app_data["secret-id"] == out.secrets[0].id
+
+
+def test_pebble_ready_web(ctx):
+    ## Arrange.
+    domain = "aotearoa.dev"
+    media = "local://"
+    user = "USER"
+    password = "PASS"
+    endpoint = "db.local:8000"
+    secret = scenario.Secret(
+        "1234", {0: {"takahe-secret-key": "old-secret"}, 1: {"takahe-secret-key": "friend"}}
+    )
+    web_container = scenario.Container(
+        "takahe-web",
+        can_connect=True,
+    )
+    storage = scenario.Storage("local-media")
+    peer_relation = scenario.PeerRelation(
+        "takahe-peer", "takahe-peer", local_app_data={"secret-id": secret.id}
+    )
+    database_relation = scenario.Relation(
+        "database", "postgresql_client", remote_app_data={"endpoints": endpoint}
+    )
+    # SCENARIO-NOTE: It would be nice if the charm test didn't need to know the
+    # layout, since the lib is hiding all of that - but maybe that's really an
+    # issue with how the lib hides everything.
+    db_secret = scenario.Secret(
+        "1235",
+        {0: {"username": user, "password": password}},
+        label=f"database.{database_relation.relation_id}.user.secret",
+    )
+    state = scenario.State(
+        leader=True,
+        relations=[peer_relation, database_relation],
+        containers=[web_container],
+        storage=[storage],
+        secrets=[secret, db_secret],
+        config={"main-domain": domain, "media-uri": media},
+    )
+
+    ## Act.
+    out = ctx.run(web_container.pebble_ready_event, state)
+
+    ## Assert.
+    cmd = (
+        "gunicorn takahe.wsgi:application -b 0.0.0.0:8001 "
+        "--access-logfile access.log --error-logfile error.log"
+    )
+    assert_running_state(
+        ctx,
+        out,
+        out.containers[0],
+        user,
+        password,
+        endpoint,
+        secret.contents[1]["takahe-secret-key"],
+        domain,
+        media,
+        "Takahē web server",
+        cmd,
+        web_container.name,
+    )
```

Comment:
IMO this is not clear, and we should at least use keyword arguments when call this function.

However, what about a different approach where the helper only builds the expected output, and the asserts are still done here. Something like:

```python
expected_plan = get_expected_plan(
    ctx=ctx,
    user=user,
    password=password,
    ...
)
assert container.plan.to_dict() == expected_plan.to_dict()
assert container.service_status[service_name] == ops.pebble.ServiceStatus.ACTIVE
```